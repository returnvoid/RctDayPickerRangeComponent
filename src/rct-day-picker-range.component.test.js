import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'mobx-react'
import { createShallow } from 'material-ui/test-utils'
import initStoryshots from '@storybook/addon-storyshots'

// import xyzStore from '@4geit/rct-xyz-store'

import RctDayPickerRangeComponent from './rct-day-picker-range.component'

const stores = {
  // xyzStore,
}

if (!process.env.DISABLE_STORYSHOTS) {
  initStoryshots({
    storyKindRegex: /^RctDayPickerRangeComponent$/,
  })
}

const shallow = createShallow({ untilSelector: 'div' })

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <Provider {...stores} >
      <RctDayPickerRangeComponent />
    </Provider>,
    div,
  )
})
// it('renders correctly', () => {
//   const wrapper = mount(<RctDayPickerRangeComponent {...stores} />)
//   expect(wrapper).toMatchSnapshot()
// })
it('shallow-renders correctly', () => {
  const wrapper = shallow(<RctDayPickerRangeComponent {...stores} />)
  expect(wrapper).toMatchSnapshot()
})
