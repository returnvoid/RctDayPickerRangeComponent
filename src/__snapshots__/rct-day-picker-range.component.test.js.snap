// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow-renders correctly 1`] = `
<div>
  <WithStyles(Button)
    onClick={[Function]}
  >
    <WithStyles(GridWrapper)
      alignItems="center"
      container={true}
      justify="space-between"
    >
      <WithStyles(GridWrapper)
        item={true}
      >
        <WithStyles(GridWrapper)
          alignItems="center"
          container={true}
        >
          <WithStyles(GridWrapper)
            item={true}
          >
            <WithStyles(Typography)>
              Dates
            </WithStyles(Typography)>
          </WithStyles(GridWrapper)>
        </WithStyles(GridWrapper)>
      </WithStyles(GridWrapper)>
      <WithStyles(GridWrapper)
        item={true}
      >
        <WithStyles(Icon)
          className="WithTheme-WithWidth-RctDayPickerRangeComponent---icon-1"
        >
          keyboard_arrow_down
        </WithStyles(Icon)>
      </WithStyles(GridWrapper)>
    </WithStyles(GridWrapper)>
  </WithStyles(Button)>
  <WithStyles(Popover)
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
    anchorPosition={
      Object {
        "left": 400,
        "top": 200,
      }
    }
    anchorReference="anchorEl"
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <DayPickerRangeController
      daySize={39}
      enableOutsideDays={false}
      endDate={null}
      firstDayOfWeek={null}
      focusedInput="startDate"
      hideKeyboardShortcutsPanel={false}
      initialVisibleMonth={null}
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isFocused={false}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      minimumNights={1}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={2}
      onBlur={[Function]}
      onClose={[Function]}
      onDatesChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onOutsideClick={[Function]}
      onPrevMonthClick={[Function]}
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "chooseAvailableEndDate": [Function],
          "chooseAvailableStartDate": [Function],
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonth={null}
      showKeyboardShortcuts={false}
      startDate={null}
      verticalHeight={null}
      weekDayFormat="dd"
      withPortal={false}
    />
  </WithStyles(Popover)>
</div>
`;
